src/Cardano/Shell/Lib.hs

```diff
@@ -147,7 +149,7 @@ runCardanoApplicationWithFeatures _ cardanoFeatures cardanoApplication = do
         pure ()
 
 
-type AllFeaturesInitFunction = CardanoConfiguration -> CardanoEnvironment -> IO [CardanoFeature]
+type AllFeaturesInitFunction = PartialCardanoConfiguration -> CardanoEnvironment -> IO [CardanoFeature]
```

Comment:
This is kind of weird. What is the point of `CardanoConfiguration` then?

Reply:
The idea is that `Partial*` variants are:

- allowed to be incomplete
- mergeable

So the `finalise*` family functions serve as a decision point where we say "ok, we merged all sources of configuration, and the the result is fully defined and ready for use".